core/src/test/scala/org/apache/spark/resource/ResourceProfileManagerSuite.scala

```diff
@@ -98,6 +98,36 @@ class ResourceProfileManagerSuite extends SparkFunSuite {
     assert(error.contains("ResourceProfiles are only supported on YARN with dynamic allocation"))
   }
 
+  test("ResourceProfileManager has equivalent profile") {
+    val conf = new SparkConf().set(EXECUTOR_CORES, 4)
+    val rpmanager = new ResourceProfileManager(conf)
+    var rpAlreadyExist: Option[ResourceProfile] = None
+    val checkId = 500
+    for (i <- 1 to 1000) {
+      val rprofBuilder = new ResourceProfileBuilder()
+      val ereqs = new ExecutorResourceRequests()
+      ereqs.cores(i).memory("4g").memoryOverhead("2000m")
+      val treqs = new TaskResourceRequests()
+      treqs.cpus(i)
```

Comment:
It's not related to this PR, but just wondering why we use different API name between executor and task for cpu/cores. Maybe it's follow original confs?

Reply:
yes, that is the way the spark confs are defined. task config is called spark.task.cpus for some reason. I have no idea why, its always been that way, but wanted to follow that convention